src/recensio/plone/behaviors/base.py

```diff
@@ -0,0 +1,234 @@
+
+
+# from plone.supermodel.directives import fieldset
+
+
+# TODO, maybe:
+# - Move `title` to fieldset "reviewed_text"
+# - Move `subject` to fieldset "reviewed_text"
+# - Set content type for `generatedPdf` to `application/pdf`
+# - Set a character limit for `review`: 4000 characters
+# - Set rows for `review` to 20
+
+# TODO: REVIEW the following.
+# Left out attributes from recensio.contenttype.content.schemata.BaseReviewSchema:
+# - reviewAuthorLastname: For migration to reviewAuthors only
+# - reviewAuthorFirstname: For migration to reviewAuthors only
+# - uri: Partner URL is no longer used for reviews but was being kept to
+#        avoid breakage
+#        this is still used in presentations. #3103
+
+
+@provider(IFormFieldProvider)
+class IBase(model.Schema):
+    directives.widget(
+        "reviewAuthors",
+        RelatedItemsFieldWidget,
+        pattern_options={"mode": "auto", "favorites": []},
+    )
+    reviewAuthors = RelationList(
+        title=_("label_review_authors"),
+        defaultFactory=list,
+        value_type=RelationChoice(source=CatalogSource(portal_type="Person")),
+        required=True,
+    )
+
+    directives.widget("languageReview", SelectFieldWidget)
+    languageReview = schema.Choice(
+        title=_("Language(s) (review)"),
+        vocabulary="plone.app.vocabularies.SupportedContentLanguages",
+        required=False,
+        missing_value="",
+        defaultFactory=default_language,
+    )
+
+    directives.widget("languageReviewedText", SelectFieldWidget)
+    languageReviewedText = schema.Choice(
+        title=_("Language(s) (text)"),
+        vocabulary="plone.app.vocabularies.SupportedContentLanguages",
+        required=False,
+        missing_value="",
+        defaultFactory=default_language,
+    )
+
+    recensioID = schema.TextLine(
+        required=False,
+    )
+    directives.omitted("recensioID")
+
+    generatedPdf = namedfile.NamedBlobFile(
+        title=_("Generated Pdf"),
+        required=False,
+    )
+    directives.omitted("generatedPdf")
+    directives.no_omit(IEditForm, "generatedPdf")
+    directives.no_omit(IAddForm, "generatedPdf")
+
+    directives.widget("review", RichTextFieldWidget)
+    searchable("review")
+    review = RichTextField(
+        title=_("Core Statements"),
+        required=False,
+        allowed_mime_types=["text/html"],
+    )
+
+    urn = schema.TextLine(
+        title=_("URN"),
+        description=_("description_urn", default="Filled in by the editorial staff"),
+        required=False,
+    )
+
+    bv = schema.TextLine(
+        title=_("BV Number"),
+        description=_(
+            "description_bv_number", default="Filled in by the editorial staff"
+        ),
+        required=False,
+    )
+
+    ppn = schema.TextLine(
+        title=_("PPN"),
+        description=_(
+            "description_bv_number", default="Filled in by the editorial staff"
+        ),
+        required=False,
+    )
+
+    canonical_uri = schema.URI(
```

Comment:
This is an `atapi.StringField` in the old code. Could there be issues when changing to `schema.URI`? If I'm not mistaken the validators differ (old recensio does not require a scheme).